L/LibGit2/bundled/patches/libgit2-agent-nonfatal.patch

```diff
@@ -1,25 +0,0 @@
-commit 70020247d1903c7a1262d967cf205a44dc6f6ebe
-Author: Keno Fischer <kfischer@college.harvard.edu>
-Date:   Wed Jul 20 19:59:00 2016 -0400
-
-    Make failure to connect to ssh-agent non-fatal
-
-    Julia issue: https://github.com/JuliaLang/julia/pull/17459
-    Upstream: https://github.com/libgit2/libgit2/issues/3866
-
-diff --git a/src/transports/ssh.c b/src/transports/ssh.c
-index cfd5736..82d2c63 100644
---- a/src/transports/ssh.c
-+++ b/src/transports/ssh.c
-@@ -296,8 +296,10 @@ static int ssh_agent_auth(LIBSSH2_SESSION *session, git_cred_ssh_key *c) {
- 
- 	rc = libssh2_agent_connect(agent);
- 
--	if (rc != LIBSSH2_ERROR_NONE)
-+	if (rc != LIBSSH2_ERROR_NONE) {
-+		rc = LIBSSH2_ERROR_AUTHENTICATION_FAILED;
- 		goto shutdown;
-+	}
- 
```

Comment:
This patch is still needed AFAICT: https://github.com/libgit2/libgit2/blob/v1.2.0/src/transports/ssh.c#L304-L309
Also the upstream issue isn't closed: https://github.com/libgit2/libgit2/issues/3866

Reply:
Dangerous on my part to pretend to know what is going on in C code, but why wouldn't the same behavior be captured here:

https://github.com/libgit2/libgit2/blob/v1.2.0/src/transports/ssh.c#L314:L332

that's why I thought it might be redundant...

Reply:
Fair enough. Do you know whether we can get someone to look at this patch and see whether it might be irrelevant (original issue is from 5 years ago)? Seems preferable to dragging this forward until the end of time?

Reply:
Especially as according to this comment (from a separate, but related issue) from 2 years ago (3 years after the issue was opened), there was a significant refactor of the error handling code...

> We've significantly refactored the network code and w/r/t authentication; you should now only get a second request for credentials if the first ones were actually refused. (Credentials persist through redirects and other replays.) Closing as obsolete.

https://github.com/libgit2/libgit2/issues/3761#issuecomment-500854417

Reply:
Hard to tell given how old is this patch. Can you confirm whether all tests pass without it (when applying the patches to a clean Julia git clone and building with `USE_BINARYBUILDER_LIBGIT2=0`.

@wildart @Keno Do you think https://github.com/JuliaLang/julia/pull/17459 is fixed upstream now? Is it enough to run the testsuite to check that?

Reply:
I'm just noting that the conda-forge feedstock for Julia currently uses generic libgit2 and mbedtls (e.g. `USE_SYSTEM_LIBGIT2=1 USE_SYSTEM_MBEDTLS=1`) on Linux only. There seem to be more complex issues on macOS.
https://github.com/conda-forge/julia-feedstock/blob/421845a3ca31c24beef378b78e7fe5209e7b871b/recipe/build.sh#L25